include/net/lwm2m.h

```diff
@@ -1062,6 +1062,22 @@ int lwm2m_engine_update_service_period(k_work_handler_t service, uint32_t period
  */
 int lwm2m_update_device_service_period(uint32_t period_ms);
 
+/**
+ * @brief Check whether a path is observed
+ *
+ * @param[in] pathstr LwM2M path string to check, e.g. "3/0/1"
+ *                    Note: level 4 (resource-instance) is treated the exact same
+ *                    as level 3 (resource) since there exist no resource instance
+ *                    specific observations in LwM2M.
+ *
+ * @return 1 when there exists an observation of the same level
+ *         or lower as the given path, 0 otherwise. Negative if the given string
+ *         is not a valid LwM2M-path.
+ *         E.g. true if path refers to a resource and the parent object has an
+ *         observation, false for the inverse.
```

Comment:
I don't think we should use `true`/`false` in this regard, `-1` could also be interpreted as `true`.

Reply:
I'm leaning towards having this function returning just booleans. `true` when observed, `false` if not, or if the path is wrong. Paths are mostly hard coded anyway, so eating the error code is so trivial that it will be noticed in the first test.

Reply:
Actually that is a good idea, +1 to stick to booleans only.

Reply:
Yeah, initially I had it return booleans but then changed my mind midway because of the possibility of the parsing of the path to fail and clearly I wasn't consistent when changing it :sweat_smile:
I also like the approach of just returning false when the path is invalid, changed.